source/language/types.rst

```diff
@@ -162,27 +162,62 @@ vendors may not support manipulating these values at run-time.
    // Declare a single-precision 32-bit float
    float[32] my_float = π;
 
-Fixed-point angles
-~~~~~~~~~~~~~~~~~~
-
-Fixed-point angles are interpreted as 2π times a 0:0:n
-fixed-point number. This represents angles in the interval
-:math:`[0,2\pi)` up to an error :math:`\epsilon\leq \pi/2^{n-1}` modulo
-2π. The statement ``angle[size] name;`` declares an n-bit angle. OpenQASM3
-introduces this specialized type because of the ubiquity of this angle
-representation in phase estimation circuits and numerically controlled
-oscillators found in hardware platform. Note that defining gate
-parameters with ``angle`` types may be necessary for those parameters to be
-compatible with run-time values on some platforms.
+.. _angle-type:
 
-.. code-block:: c
-   :force:
+Angles
+~~~~~~
 
-   // Declare an angle with 20 bits of precision and assign it a value of π/2
+OpenQASM 3 includes a new type to represent classical angles: ``angle``.
+This type is intended to make manipulations of angles more efficient at runtime,
+when the hardware executing the program does not have built-in support for
+IEEE-754 floating-point.  The manipulations on ``angle`` values are designed to
+be significantly less expensive when done using integer hardware than the
+equivalent software emulation of IEEE-754 operations.
+
+In brief, the type ``angle[size]`` is manipulated very similarly to a single
+unsigned integer, where the value ``1`` represents an angle of
+:math:`2\pi/2^{-\text{size}}`, and the largest representable value is
+this subtracted from :math:`2\pi`.  Addition with other angles, and
+multiplication and division by unsigned integers is defined by standard
+unsigned-integer arithmetic, with more details found in :ref:`the section on
+classical instructions <classical-instructions>`.
+
+The statement ``angle[size] name;`` statement declares a new angle called
+``name`` with ``size`` bits in its representation.  Angles can be assigned
+values using the constant ``π`` or ``pi``, such as::
+
+   // Declare a 20-bit angle with the value of "π/2"
    angle[20] my_angle = π / 2;
-   float[32] float_pi = π;
-   // equivalent to pi_by_2 up to rounding errors
-   angle[20](float_pi / 2);
+
+The bit representation of the type ``angle[size]`` is such that if
+``angle_as_uint`` is the integer whose representation as a ``uint[size]`` has
+the same bit pattern, the value of the angle (using exact mathematical
+operations on the field of real numbers) would be
+
+.. math::
+
+   2\pi \times \frac{\text{angle_as_uint}}{2^{\text{size}}}
+
+This "mathematical" value is the value used in casts from floating-point values
+(if available), whereas casts to and from ``bit[size]`` types reinterpret the
+bits directly.  This means that, unless ``a`` is sufficiently small::
+
+  float[32] a;
+  angle[32](bit[32](uint[32](a))) != angle[32](a)
```

Comment:
**question**: I'm confused about the role `bit` is supposed to play here. Is this a point about losing precision or something else?

Reply:
It's because we don't have defined casting behaviour between `uint[32]` and `angle[32]` directly, but we do if you go via `bit[32]` (which is a reinterpreting cast in both cases).